src/modules/item/item.controller.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  ParseUUIDPipe,
+  Post,
+  Put,
+  Query,
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+import {
+  ApiBearerAuth,
+  ApiOkResponse,
+  ApiOperation,
+  ApiTags,
+} from '@nestjs/swagger';
+import { Request } from 'express';
+import { JwtAuthGuard } from '../auth/jwt-auth.guard';
+import { Item } from '../../../entities/Item.entity';
+import { ItemService } from './item.service';
+import { User } from '../../../entities/User.entity';
+import { ParseOrderByPipeUsers, ParseOrderPipeUsers } from '../user/user.pipes';
+import { PageResponse } from '../../constants/PageResponse';
+import { DeleteResult } from 'typeorm';
+
+@Controller('/item')
+@ApiTags('Items')
+@ApiBearerAuth()
+@UseGuards(JwtAuthGuard)
+export class ItemController {
+  constructor(private readonly itemService: ItemService) {}
+
+  @Post('')
+  @ApiOperation({ summary: 'Upsert an Item' })
+  @ApiOkResponse({ description: 'Upsert an Item', type: Item })
+  async createItem(@Body() item: Item): Promise<Item> {
+    return this.itemService.createItem(item);
```

Comment:
Save the companyId from the session instead of item request body, this way we guarante no company can create items for other companies.

Reply:
🆗